### ISSUE TITLE
capitalize all classes in Morebits?

### ISSUE THREAD

**Issue by NovemLinguae** [azatoth/twinkle#2084]
Morebits has a bunch of ES5-style newable classes that do not begin with a capital letter, which is the normal convention for classes. This is triggering the eslint linter error `new-cap`.

This ticket is to decide whether these are worth capitalizing or not. So for example, changing `Morebits.date` to `Morebits.Date`.

---

Example Morebits code (before changing):

```js
/**
 * @class
 */
Morebits.date = function() {
	this.a = 'test';
}

Morebits.date.prototype = {
	/** @returns {boolean} */
	isValid: function() {
		return !isNaN(this.getTime());
	},

	/**
	 * @param {(Date|Morebits.date)} date
	 * @returns {boolean}
	 */
	isBefore: function(date) {
		return this.getTime() < date.getTime();
	}
}
```

Example calling code (before changing):

```js
const date = new Morebits.date(pageobj.getLoadTime());
```

---

* Option 1 - Set `"new-cap": "off"`
* Option 2 - Capitalize all Morebits classes, and add backwards compatibility code so that nothing breaks in third party libraries.

---

Plan if we decide to implement option 2 (probably spread across 4 patches):

* [x] Add backwards compatibility code towards the bottom of Morebits.js
```js
Morebits.QuickForm = Morebits.quickForm;
Morebits.QuickForm.Element = Morebits.quickForm.element;
Morebits.Unbinder = Morebits.unbinder;
Morebits.Date = Morebits.date;
Morebits.wiki.Api = Morebits.wiki.api;
Morebits.wiki.Page = Morebits.wiki.page;
Morebits.wiki.Preview = Morebits.wiki.preview;
Morebits.wikitext.Page = Morebits.wikitext.page;
Morebits.UserspaceLogger = Morebits.userspaceLogger;
Morebits.Status = Morebits.status;
Morebits.BatchOperation = Morebits.batchOperation;
Morebits.TaskManager = Morebits.taskManager;
Morebits.SimpleWindow = Morebits.simpleWindow;
```
* [x] Fix capitalization in twinkle*.js files.
* [x] Fix capitalization in tests
* [ ] Fix capitalization in Morebits.js, then invert the backwards compatibility code.

**Comment by NovemLinguae** [comment]
@siddharthvp, would appreciate your thoughts if you get a moment. I think you've done a lot of work on the Morebits library.

**Comment by siddharthvp** [comment]
Yep, let's do this. Morebits classes weren't originally following the convention, and even the new classes didn't follow it as I (and I guess Amory) didn't know the convention.
